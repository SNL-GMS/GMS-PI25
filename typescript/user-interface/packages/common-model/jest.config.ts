import type { Config } from 'jest';

const config: Config = {
  bail: true,
  cacheDirectory: '<rootDir>/.cache/jest',
  testEnvironmentOptions: {
    url: 'http://localhost/'
  },
  globalSetup: '<rootDir>/jest.global-setup.ts',
  setupFiles: ['<rootDir>/jest.setup.ts'],
  transform: {
    '^.+\\.jsx?$': 'babel-jest',
    '^.+\\.tsx?$': [
      'ts-jest',
      {
        isolatedModules: true,
        diagnostics: false,
        tsconfig: '<rootDir>/tsconfig-test.json'
      }
    ]
  },
  modulePathIgnorePatterns: ['<rootDir>/lib/', '<rootDir>/dist/'],
  testRegex: '/__tests__/.*\\.test\\.(ts|tsx)$',
  moduleFileExtensions: ['ts', 'tsx', 'js', 'json'],
  moduleNameMapper: {
    '@gms/((?!golden-layout)[^/]+)$': '<rootDir>/../$1/src/ts/$1',
    '@gms/((?!golden-layout)[^/]+)(/lib/)(.*)$': '<rootDir>/../$1/src/ts/$3',
    '.*\\.(css|less|styl|scss|sass)$': '<rootDir>/__mocks__/style-mock.ts',
    '.*\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
      '<rootDir>/__mocks__/file-mock.ts'
  },
  testEnvironment: 'node',
  collectCoverage: true,
  coverageReporters: ['lcov', 'html', 'text-summary']
};
// eslint-disable-next-line import/no-default-export
export default config;
