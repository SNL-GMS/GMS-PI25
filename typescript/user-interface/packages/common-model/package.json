{
  "name": "@gms/common-model",
  "version": "0.1.0",
  "description": "GMS Common Model",
  "license": "UNLICENSED",
  "author": "GMS Development Team",
  "sideEffects": false,
  "module": "./lib/common-model.js",
  "typings": "./lib/common-model.d.ts",
  "files": [
    "lib"
  ],
  "scripts": {
    "clean:build": "yarn rimraf lib/ dist/ docs/ coverage/ bundle-analyzer/ artifacts/ .cache/ junit.xml",
    "clean": "npm-run-all -s clean:build",
    "build:eslint": "yarn eslint -c .eslintrc-config.yaml 'src/**/*.{ts,tsx}' --quiet --cache --cache-strategy content --cache-file .cache/.eslintcache",
    "build:eslint:test": "yarn eslint -c .eslintrc-config-test.yaml '{__tests__,__mocks__,__integration-tests__}/**/*.{ts,tsx}' --quiet --cache --cache-strategy content --cache-file .cache/.eslintcache-test",
    "build": "yarn tsc -b ./tsconfig-build.json",
    "build:test": "yarn tsc -b ./tsconfig-test.json",
    "watch": "yarn tsc -b ./tsconfig-build.json -watch",
    "docs": "yarn clean:docs && typedoc --out docs ./src/ --name common-model --mode file --readme README.md --target ES6 --includeDeclarations --excludeExternals --excludeNotExported --hideGenerator",
    "sonar": "echo \"run 'yarn sonar' from the top-level directory\"",
    "test": "yarn test:dev",
    "test:dev:jest": "jest --maxWorkers=50% --verbose --bail",
    "test:prod:jest": "jest --maxWorkers=4 --silent --bail --coverageReporters=lcovonly --reporters=default --reporters=jest-junit",
    "test:dev": "npm-run-all -p build:test test:dev:jest",
    "test:prod": "npm-run-all -p build:test test:prod:jest",
    "version": "echo $npm_package_version"
  },
  "devDependencies": {
    "@blueprintjs/eslint-plugin": "2.2.0",
    "@testing-library/jest-dom": "5.16.5",
    "@testing-library/react": "13.4.0",
    "@types/faker": "4.1.8",
    "@types/jest": "29.1.1",
    "@types/jquery": "3.3.31",
    "@types/node": "18.11.9",
    "@types/react": "18.0.26",
    "@types/react-dom": "18.0.9",
    "@typescript-eslint/eslint-plugin": "5.33.0",
    "@typescript-eslint/parser": "5.33.0",
    "eslint": "8.22.0",
    "eslint-config-airbnb": "19.0.4",
    "eslint-config-prettier": "8.5.0",
    "eslint-import-resolver-typescript": "3.4.1",
    "eslint-plugin-import": "2.26.0",
    "eslint-plugin-jest": "27.0.4",
    "eslint-plugin-jsdoc": "39.3.6",
    "eslint-plugin-jsx-a11y": "6.6.1",
    "eslint-plugin-no-null": "1.0.2",
    "eslint-plugin-prefer-arrow": "1.2.3",
    "eslint-plugin-prettier": "4.2.1",
    "eslint-plugin-react": "7.30.1",
    "eslint-plugin-react-hooks": "4.6.0",
    "eslint-plugin-simple-import-sort": "7.0.0",
    "eslint-plugin-unicorn": "43.0.2",
    "eslint-plugin-workspaces": "0.7.0",
    "faker": "4.1.0",
    "jest": "29.1.2",
    "jest-canvas-mock": "2.4.0",
    "jest-cli": "29.1.2",
    "jest-environment-jsdom": "29.1.2",
    "jest-fetch-mock": "3.0.3",
    "jest-junit": "12.2.0",
    "jsdom": "20.0.1",
    "jsdom-global": "3.0.2",
    "lodash": "4.17.21",
    "node-fetch": "2.6.0",
    "npm-run-all": "4.1.5",
    "prettier": "2.1.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "3.0.0",
    "ts-jest": "29.1.1",
    "ts-node": "10.9.1",
    "typedoc": "0.15.3",
    "typescript": "5.1.6"
  },
  "jest-junit": {
    "suiteNameTemplate": "common-model/{filename}",
    "classNameTemplate": "common-model/{filename}/{classname}",
    "titleTemplate": "{title}"
  }
}
