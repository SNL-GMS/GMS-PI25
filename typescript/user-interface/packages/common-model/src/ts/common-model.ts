import * as BeamformingTemplateTypes from './beamforming-templates/types';
import * as ChannelSegmentTypes from './channel-segment/types';
import * as ColorTypes from './color/types';
import * as CommonTypes from './common/types';
import * as DataAcquisitionTypes from './data-acquisition/types';
import * as Displays from './displays/types';
import * as Endpoints from './endpoints/types';
import * as EventTypes from './event';
// required for legacy components
import * as LegacyEventTypes from './event/legacy';
import * as FacetedTypes from './faceted';
import * as FilterUtil from './filter/filter-util';
import * as FilterTypes from './filter/types';
import * as FkTypes from './fk';
import * as ProcessingMaskDefinitionTypes from './processing-mask-definitions/types';
// required for legacy components
import * as QcMaskTypes from './qc-mask';
import * as QcSegmentTypes from './qc-segment';
import * as SignalDetectionTypes from './signal-detection';
import * as SohTypes from './soh/types';
import * as ChannelTypes from './station-definitions/channel-definitions/channel-definitions';
import * as ResponseTypes from './station-definitions/channel-definitions/response-definitions';
import * as StationTypes from './station-definitions/station-definitions/station-definitions';
import * as ProcessingStationTypes from './station-processing/types';
import * as ReferenceStationTypes from './station-reference/types';
import * as SystemMessageTypes from './system-message/types';
import * as TimeTypes from './time';
import * as TypeUtil from './type-util/type-util';
import * as ConfigurationTypes from './ui-configuration/types';
import * as UserProfileTypes from './user-profile/types';
import * as WaveformTypes from './waveform/types';
import * as WorkflowTypes from './workflow/types';

export {
  BeamformingTemplateTypes,
  ChannelSegmentTypes,
  ChannelTypes,
  ColorTypes,
  CommonTypes,
  ConfigurationTypes,
  DataAcquisitionTypes,
  Displays,
  Endpoints,
  EventTypes,
  FacetedTypes,
  FilterTypes,
  FilterUtil,
  FkTypes,
  LegacyEventTypes,
  ProcessingMaskDefinitionTypes,
  ProcessingStationTypes,
  QcMaskTypes,
  QcSegmentTypes,
  ReferenceStationTypes,
  ResponseTypes,
  SignalDetectionTypes,
  SohTypes,
  StationTypes,
  SystemMessageTypes,
  TimeTypes,
  TypeUtil,
  UserProfileTypes,
  WaveformTypes,
  WorkflowTypes
};
