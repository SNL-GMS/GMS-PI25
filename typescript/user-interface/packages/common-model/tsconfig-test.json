{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "tsBuildInfoFile": "./.cache/.tsbuildinfo-test",
    "paths": {}
  },
  "references": []
}
