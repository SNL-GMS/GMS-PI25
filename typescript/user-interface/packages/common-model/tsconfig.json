{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./lib"
  },
  "include": ["src/**/*", "__tests__/**/*", "__mocks__/**/*"]
}
