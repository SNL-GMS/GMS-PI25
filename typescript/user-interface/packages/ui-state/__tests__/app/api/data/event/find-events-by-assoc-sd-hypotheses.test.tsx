import { CommonTypes, SignalDetectionTypes } from '@gms/common-model';
import type { AnyAction } from 'redux';
import type { MockStoreCreator } from 'redux-mock-store';
import createMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { AppState } from '../../../../../src/ts/app';
import {
  addFindEventsByAssociatedSignalDetectionHypothesesReducers,
  findEventsByAssociatedSignalDetectionHypotheses,
  shouldSkipFindEventsByAssociatedSignalDetectionHypotheses
} from '../../../../../src/ts/app/api/data/event/find-events-by-assoc-sd-hypotheses';
import type { FindEventsByAssociatedSignalDetectionHypothesesQueryArgs } from '../../../../../src/ts/app/api/data/event/types';
import { appState } from '../../../../test-util';

const mockArgs: FindEventsByAssociatedSignalDetectionHypothesesQueryArgs = {
  signalDetectionHypotheses: [
    {
      id: {
        id: '20cc9505-efe3-3068-b7d5-59196f37992c',
        signalDetectionId: '012de1b9-8ae3-3fd4-800d-58665c3152cc'
      },
      parentSignalDetectionHypothesis: null,
      deleted: false,
      station: {
        name: 'ASAR',
        effectiveAt: null
      },
      monitoringOrganization: 'GMS',
      featureMeasurements: [
        {
          channel: {
            name:
              'ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4',
            effectiveAt: 1636503404
          },
          measuredChannelSegment: {
            id: {
              channel: {
                name:
                  'ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4',
                effectiveAt: 1636503404
              },
              startTime: 1636503404,
              endTime: 1636503704,
              creationTime: 1636503404
            }
          },
          measurementValue: {
            arrivalTime: {
              value: 1636503404,
              standardDeviation: 1.162
            },
            travelTime: null
          },
          snr: {
            value: 8.9939442,
            standardDeviation: null,
            units: CommonTypes.Units.DECIBELS
          },
          featureMeasurementType: SignalDetectionTypes.FeatureMeasurementType.ARRIVAL_TIME
        }
      ]
    }
  ],
  stageId: { name: 'AL1' }
};

describe('Find events by associated SD hypotheses', () => {
  test('contains definitions', () => {
    expect(addFindEventsByAssociatedSignalDetectionHypothesesReducers).toBeDefined();
    expect(findEventsByAssociatedSignalDetectionHypotheses).toBeDefined();
    expect(shouldSkipFindEventsByAssociatedSignalDetectionHypotheses).toBeDefined();
  });

  test('query is skipped when args are falsy, not skipped when args are truthy', () => {
    const mockSkipArgs: FindEventsByAssociatedSignalDetectionHypothesesQueryArgs = {
      signalDetectionHypotheses: [],
      stageId: null
    };
    expect(shouldSkipFindEventsByAssociatedSignalDetectionHypotheses(undefined)).toBeTruthy();
    expect(shouldSkipFindEventsByAssociatedSignalDetectionHypotheses(mockSkipArgs)).toBeTruthy();
    expect(shouldSkipFindEventsByAssociatedSignalDetectionHypotheses(mockArgs)).toBeFalsy();
  });

  it('will not execute query if the args are invalid', async () => {
    const mockStoreCreator: MockStoreCreator<AppState, AnyAction> = createMockStore([thunk]);

    const store = mockStoreCreator(appState);

    await store.dispatch(
      findEventsByAssociatedSignalDetectionHypotheses({
        ...mockArgs,
        stageId: null
      }) as any
    );

    // results should have empty arrays since current interval is not set
    expect(store.getActions()).toHaveLength(0);
  });

  test('findEventsByAssociatedSignalDetectionHypotheses returns events', () => {
    expect(findEventsByAssociatedSignalDetectionHypotheses(mockArgs)).toMatchSnapshot();
  });

  test('addFindEventsByAssociatedSignalDetectionHypothesesReducers builds', () => {
    const mapKeys = [
      'eventManagerApi/findEventsByAssociatedSignalDetectionHypotheses/pending',
      'eventManagerApi/findEventsByAssociatedSignalDetectionHypotheses/fulfilled',
      'eventManagerApi/findEventsByAssociatedSignalDetectionHypotheses/rejected'
    ];
    const builderMap = new Map();
    const builder: any = {
      addCase: (k, v) => {
        builderMap.set(k.type, v);
        return builder;
      }
    };
    addFindEventsByAssociatedSignalDetectionHypothesesReducers(builder);
    expect(builderMap).toMatchSnapshot();
    // eslint-disable-next-line prefer-const
    const state = { queries: { findEventsByAssociatedSignalDetectionHypotheses: {} } };
    // eslint-disable-next-line prefer-const
    let action = {
      meta: { requestId: 12345, mockArgs },
      payload: { events: [] }
    };

    builderMap.get(mapKeys[0])(state, action);
    expect(state).toMatchSnapshot();
    builderMap.get(mapKeys[1])(state, action);
    expect(state).toMatchSnapshot();
    builderMap.get(mapKeys[2])(state, action);
    expect(state).toMatchSnapshot();
  });
});
