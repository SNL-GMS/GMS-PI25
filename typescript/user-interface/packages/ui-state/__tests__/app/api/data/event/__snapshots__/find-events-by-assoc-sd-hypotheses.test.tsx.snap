// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find events by associated SD hypotheses addFindEventsByAssociatedSignalDetectionHypothesesReducers builds 1`] = `
Map {
  "eventManagerApi/findEventsByAssociatedSignalDetectionHypotheses/pending" => [Function],
  "eventManagerApi/findEventsByAssociatedSignalDetectionHypotheses/fulfilled" => [Function],
  "eventManagerApi/findEventsByAssociatedSignalDetectionHypotheses/rejected" => [Function],
}
`;

exports[`Find events by associated SD hypotheses addFindEventsByAssociatedSignalDetectionHypothesesReducers builds 2`] = `
{
  "queries": {
    "findEventsByAssociatedSignalDetectionHypotheses": {
      "eventsByAssociatedSignalDetectionHypotheses": {
        "12345": {
          "arg": undefined,
          "error": undefined,
          "status": "pending",
        },
      },
    },
  },
}
`;

exports[`Find events by associated SD hypotheses addFindEventsByAssociatedSignalDetectionHypothesesReducers builds 3`] = `
{
  "queries": {
    "findEventsByAssociatedSignalDetectionHypotheses": {
      "eventsByAssociatedSignalDetectionHypotheses": {
        "12345": {
          "arg": undefined,
          "error": undefined,
          "status": "fulfilled",
        },
      },
    },
  },
}
`;

exports[`Find events by associated SD hypotheses addFindEventsByAssociatedSignalDetectionHypothesesReducers builds 4`] = `
{
  "queries": {
    "findEventsByAssociatedSignalDetectionHypotheses": {
      "eventsByAssociatedSignalDetectionHypotheses": {
        "12345": {
          "arg": undefined,
          "error": undefined,
          "status": "rejected",
        },
      },
    },
  },
}
`;

exports[`Find events by associated SD hypotheses findEventsByAssociatedSignalDetectionHypotheses returns events 1`] = `[Function]`;
